src/base/QXmppJingleIq.cpp

```diff
@@ -497,7 +496,9 @@ bool QXmppJingleIq::Content::parseSdp(const QString &sdp)
                         if (payload.name() == QStringLiteral("telephone-event")) {
                             params.insert(QStringLiteral("events"), paramStr);
                         } else {
-                            for (const auto &p : paramStr.split(QRegularExpression(";\\s*"))) {
+                            thread_local static const auto regex = QRegularExpression(";\\s*");
```

Comment:
Does it need to be `thread_local`?

Reply:
QRegularExpression at least doesn't say it's thread-safe and from what I read (https://stackoverflow.com/questions/25581696/qregularexpression-matching-thread-safety ) that's probably required.

Reply:
Interesting.

Let's play it safe here then, the overhead of `thread_local` is insignificant compared to the rest of what the code does, let alone the networking involved here.